source/sigv4.c

```diff
@@ -1972,53 +2021,50 @@ static void generateCredentialScope( const SigV4Parameters_t * pSigV4Params,
                                       pCanonicalRequest->pQueryLoc[ paramsIndex ].key.dataLen,
                                       pBufLoc,
                                       &encodedLen,
-                                      true,
-                                      false );
+                                      true /* Encode slash (/) */,
+                                      false /* Do not encode '='. */ );
 
             if( returnStatus == SigV4Success )
             {
                 pBufLoc += encodedLen;
                 remainingLen -= encodedLen;
 
-                /* Encode parameter value if non-empty. Query parameters can have empty values. */
-                if( pCanonicalRequest->pQueryLoc[ paramsIndex ].value.dataLen > 0U )
```

Comment:
Why can this be assumed to be true now? Or does `writeValueInCanonicalizedQueryString` still work with a zero length parameter?

Reply:
>Or does writeValueInCanonicalizedQueryString still work with a zero length parameter?

Yes, this function adds the `=` character which is required for canonical query format even with empty values.